examples/multi-threaded/main.c

```diff
@@ -1,19 +1,26 @@
-// Copyright (c) 2020 Cesanta Software Limited
+// Copyright (c) 2020-2023 Cesanta Software Limited
 // All rights reserved
 //
 // Multithreading example.
 // For each incoming request, we spawn a separate thread, that sleeps for
 // some time to simulate long processing time, produces an output and
 // hands over that output to the request handler function.
 //
+// We use a single queue for the bi-directional communication of the connection
+// and a worker thread. A connection starts a worker thread and writes a string
+// value (a name) into the queue. A worker thread reads a name from a queue,
+// and writes a string message "Hello, <name>" back.  A connection waits for
+// that message to appear, and sends it back to the client.
+//
+// The 1st byte of the message in the queue specifies the direction.
```

Comment:
This works because there is always only one outstanding message in the queue...

Reply:
Yes. That works only in the "dialogue" mode.

Reply:
I would add a note to avoid github issues accumulating